making_change/making_change.py

```diff
@@ -1,16 +1,34 @@
 #!/usr/bin/python
 
 import sys
-
+# 1 * 0 + 5 * 0 = 0
+# [1, 5]
+#   making_change(100, [1, 5]):
+#     making_change(100 -1,[1, 5])
+#       making_change(100,[5])
 def making_change(amount, denominations):
-  pass 
+  # amount cents 
+  # demoniations lists of different coins
+  # iterate through demonination  
+  # return number of combinations
+  # storage_list to keep track of all of the ways 
+  # storage amount + 1
+  if amount == 0:
+    return 1 # 0 pennies and nickels etc
+  if amount < 0:
+    return 0
+  # if len(denominations) == 0:
+  #   return 0
+  if len(denominations) == 1:
+    return 1
+  return making_change(amount - denominations[-1], denominations) + making_change(amount, denominations[:-1])
```

Comment:
great knowing that the denominations list is sorted (because you create it) lets you take advantage of the order by merely calling the last index to check if the largest denomination works and so on and so forth. this way you need not mutate any objects and thus not have to worry about repetition errors